x/incentive/README.md

```diff
@@ -0,0 +1,301 @@
+# Incentive
+
+## Table of Contents
+
+1. [Introduction](#1-introduction)
+2. [States](#2-states)
+    1. [Parameters](#21-parameters)
+    2. [Gauge](#22-gauge)
+    3. [Reward Gauge](#23-reward-gauge)
+    4. [FinalityProviderHistoricalRewards](#24-finalityproviderhistoricalrewards)
+    5. [FinalityProviderCurrentRewards](#25-finalityprovidercurrentrewards)
+    6. [BTCDelegationRewardsTracker](#26-btcdelegationrewardstracker)
+3. [Messages](#3-messages)
+    1. [MsgUpdateParams](#31-msgupdateparams)
+    2. [MsgSetWithdrawAddress](#32-msgsetwithdrawaddress)
+    3. [MsgWithdrawReward](#33-msgwithdrawreward)
+4. [BeginBlocker](#4-beginblocker)
+5. [Queries](#5-queries)
+
+## 1. Introduction
+
+The Incentive module is responsible for determining the eligibility of rewards
+for both finality providers and BTC stakers. Initially, rewards are calculated
+based on the voting power of the finality provider, with the remaining rewards
+distributed to BTC delegations according to their voting power share.
+
+Rewards and distributions are triggered at the beginning of each
+block during the [BeginBlocker](https://docs.cosmos.network/main/build/building-modules/beginblock-endblock#beginblocker-and-endblocker-1)
+phase, ensuring immediate availability.
+
+The module is designed to manage the distribution of rewards for both BTC
+stakers and finality providers. Rewards are allocated based on the voting
+power of the finality provider and the BTC delegations. This module works
+in collaboration with the `x/distribution`. For native stakers, rewards are
```

Comment:
Can we add more description on the collaboration between x/distribution and x/incentive
I don't see that much collaboration, but the incentive is heavily inspired (almost a copy in the sense of rewards distribution) of the [AllocateTokens](https://github.com/cosmos/cosmos-sdk/blob/e76102f885b71fd6e1c1efb692052173c4b3c3a3/x/distribution/keeper/allocation.go#L17)